src/utils/collectionReducer.ts

```diff
@@ -0,0 +1,21 @@
+const collectionReducer = (collection: Collection, action: Action) => {
```

Comment:
Using a reducer here is a decent idea! Only problem is you're just doing an [idempotent update](https://restfulapi.net/idempotent-rest-apis/) by overriding the entire value each time. You could've just used a useState instead of a useReducer and had the same result with less complexity.

If you want to make your API simpler, you could stick with the useReducer, but instead of accepting the entire Collection and an action, you could accept a single card and an action. Based of what the action is (add/delete) and the card provided, you can determine how the state should update.